src/Cake/Commands/DefaultCommandSettings.cs

```diff
@@ -59,5 +59,9 @@ public sealed class DefaultCommandSettings : CommandSettings
         [CommandOption("--info")]
         [Description("Displays additional information about Cake.")]
         public bool ShowInfo { get; set; }
+
+        [CommandOption("--regeneratecache")]
```

Comment:
Use `--recompile` instead or as an alias for consistency?

Also, `--regenerate-cache` would be more in line with cli conventions; i.e. separate words with dashes.

Reply:
I had --recompile in a previous pull request and they wanted that changed at that time.  That is when I made it --regeneratecache.  If we get a consensus on what is desired I will change it.

Reply:
I'd say `--invalidate-cache`.